js/pokemon.js

```diff
@@ -357,17 +370,49 @@ function revealPokemon(correctlyGuessed, language) {
 
 function generateNewNumbers(force) {
 
-    if(force || (currentGen !== newGen)) {
-        var i=0, j;
-        
+    if(force || !sortCompareArrays(currentGen, newGen)) {
+
+        console.log("generating New Pokemon array")
+
         upcomingPokemon = new Array();
         upcomingPokemonArrayPos = 0;
-        
-        for(j=minPokemonNumber; j<=maxPokemonNumber; j++) {
-            upcomingPokemon[i] = j;
-            i++;
+        var i = 0;
+
+        for (var j=0; j < newGen.length; j++) {
```

Comment:
This for loop could be tidier - no need for all the repetition. Maybe make an object earlier in the file which defines the lower and upper bounds of each generation, and then apply them dynamically here.

```
var generations = {
1: {
   start: 1,
   end: 151
}, etc...
```

So, instead of all the if statements, you would have a single for inside here, something like this (untested):
```
 for(var k=newGen[j].start; k<=newGen[j].end; k++) {
    upcomingPokemon[i] = k;
    i++;
}
```